### PR TITLE
add custom authenticator for outbound requests to a http proxy [wip]

### PR DESCRIPTION
**Pull request by maleck13**
Reading about how to authenticate against a proxy this seemed like a good approach. Still needs testing and trying

### CODE REVIEW

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/configuration/ProxyConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+public class ProxyConfiguration {
+
+  @PostConstruct
+  public void setupAuth(){
+    System.out.println("*** setup auth called ***");
+    Authenticator.setDefault(new Authenticator() {
+
+      @Override protected PasswordAuthentication getPasswordAuthentication() {
+        System.out.println("*** getPasswordAuthentication called ***");
```
Comment (maleck13) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/configuration/ProxyConfiguration.java]: Note will remove these

**Comment by maleck13** [comment]
@matzew Believe I now have correct branches etc. Any thoughts on this?

**Comment by maleck13** [comment]
retest this please

**Comment by maleck13** [comment]
@matzew ping

**Comment by maleck13** [comment]
@wtrocki ok to take a look

**Comment by wtrocki** [comment]
@maleck13 Pull request for openshift templates was triggered but since we using some custom branch "ghprbTargetBranch: rhmap" bump failed: 
```
{
  "message": "Validation Failed",
  "errors": [
    {
      "resource": "PullRequest",
      "field": "base",
      "code": "invalid"
    }
  ]
```
base=rhmap

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java [hunk 2]
```diff
@@ -35,12 +35,7 @@
 
 import javax.inject.Inject;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (matzew) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java]: :-1:  :smile:

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java [hunk 3]
```diff
@@ -16,14 +16,7 @@
...
 
-import com.notnoop.apns.APNS;
-import com.notnoop.apns.ApnsDelegateAdapter;
-import com.notnoop.apns.ApnsNotification;
-import com.notnoop.apns.ApnsService;
-import com.notnoop.apns.ApnsServiceBuilder;
-import com.notnoop.apns.DeliveryError;
-import com.notnoop.apns.EnhancedApnsNotification;
-import com.notnoop.apns.PayloadBuilder;
+import com.notnoop.apns.*;
```
Comment (matzew) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java]: what's wrong with your IDEA ? 

we reabase thees classes from upsteam - please don't add formatting changes

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java (hunk 2 above)
Comment (maleck13) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java]: Yeah I think my editor did this by itself :(

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java (hunk 3 above)
Comment (maleck13) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java]: I think it is just naturally removing whitespace. I will revert these files

**Comment by maleck13** [comment]
@matzew  cleaned up the white space and import issues

**Pull request merged**